xarray/core/coordinates.py

```diff
@@ -107,8 +108,44 @@ def to_index(self, ordered_dims: Sequence[Hashable] = None) -> pd.Index:
             return self._data.get_index(dim)  # type: ignore
         else:
             indexes = [self._data.get_index(k) for k in ordered_dims]  # type: ignore
-            names = list(ordered_dims)
-            return pd.MultiIndex.from_product(indexes, names=names)
+
+            # compute the sizes of the repeat and tile for the cartesian product
+            # (taken from pandas.core.reshape.util)
+            lenX = np.fromiter((len(index) for index in indexes), dtype=np.intp)
+            cumprodX = np.cumproduct(lenX)
```

Comment:
could you please rename these variables to conform to the PEP8 convention? e.g., `index_lengths` and `cumprod_lengths`?

Reply:
Done. (I picked this code/these names from pandas...)